Code/Tools/AssetProcessor/native/ui/BuilderInfoMetricsModel.cpp

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include <ui/BuilderInfoMetricsModel.h>
+#include <ui/BuilderDataItem.h>
+#include <ui/BuilderData.h>
+#include <AssetBuilderSDK/AssetBuilderSDK.h>
+#include <QTime>
+
+namespace AssetProcessor
+{
+    QString DurationToQString(AZ::s64 durationInMs)
+    {
+        const AZ::s64 dayInMs = 86400000;
+        if (durationInMs < 0)
+        {
+            return "";
+        }
+
+        AZ::s64 dayCount = durationInMs / dayInMs;
+        QTime duration = QTime::fromMSecsSinceStartOfDay(durationInMs % dayInMs);
+        if (dayCount > 0)
+        {
+            return duration.toString("zzz' ms, 'ss' sec, 'mm' min, 'hh' hr, %1 day'").arg(dayCount);
+        }
+        else
+        {
+            if (!duration.isValid())
+            {
+                return "";
+            }
+            if (duration.hour() > 0)
+            {
+                return duration.toString("zzz' ms, 'ss' sec, 'mm' min, 'hh' hr'");
+            }
+            if (duration.minute() > 0)
+            {
+                return duration.toString("zzz' ms, 'ss' sec, 'mm' min'");
+            }
+            if (duration.second() > 0)
+            {
+                return duration.toString("zzz' ms, 'ss' sec'");
+            }
+            return duration.toString("zzz' ms'");
+        }
```

Comment:
you can probably drop the else block and do something like this. wrapping the duration like this shows the dependency better. requires duration to be valid to evaluate the containing statments. 
```
if (duration.isValid()) 
{
           if (duration.hour() > 0)
            {
                return duration.toString("zzz' ms, 'ss' sec, 'mm' min, 'hh' hr'");
            }
            if (duration.minute() > 0)
            {
                return duration.toString("zzz' ms, 'ss' sec, 'mm' min'");
            }
            if (duration.second() > 0)
            {
                return duration.toString("zzz' ms, 'ss' sec'");
            }
            return duration.toString("zzz' ms'");
}
return QString();
```